www/class/centreonWidget.class.php

```diff
@@ -1337,6 +1336,22 @@ public function getWidgetPreferences($widgetId)
             throw new \Exception("An error occured");
         }
 
+        //if user has no preferences take parent preferences
+        if ($this->db->numberRows() === 0) {
+            $query = 'SELECT pref.preference_value, param.parameter_code_name ' .
+                'FROM widget_preferences pref, widget_parameters param, widget_views wv ' .
+                'WHERE param.parameter_id = pref.parameter_id ' .
+                'AND pref.widget_view_id = wv.widget_view_id ' .
+                'AND wv.widget_id = :widgetId ';
+            // Prevent SQL injection with widget id
+            $stmt = $this->db->prepare($query);
+            $stmt->bindParam(':widgetId', $widgetId, \PDO::PARAM_INT);
+            $dbResult = $stmt->execute();
+            if (!$dbResult) {
```

Comment:
PDO is configured with PDO::ERRMODE_EXCEPTION flag, then it will throw an exception if query fails.
Need to wrap it in try .. catch